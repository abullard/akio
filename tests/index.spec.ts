import { describe, it, expect } from "vitest";
import { execa } from 'execa';

describe('index.ts', () => {
    it('should run test:stub npm script', async () => {
        const cmd = "pnpm";
        const args = ['akio', 'test:stub'];

        const subprocess = execa(cmd, args, {
            stdin: 'pipe'
        });

        subprocess.stdin!.write('1\n');
        subprocess.stdin!.end();

        const { stdout } = await subprocess;
        expect(stdout).toMatchSnapshot();
    });

    it('should run search for build scripts', async () => {
        const cmd = "pnpm";
        const args = ['akio', '-i', 'build'];

        const { stdout } = await execa(cmd, args, {});

        expect(stdout).toMatchSnapshot();
    });

    it('should turn formatting off', async () => {
        const cmd = "pnpm";
        const args = ['akio', '-f'];

        // TODO AJB 05/26/2025: make all of these tests normalized, and run one at a time
        const { stdout } = await execa(cmd, args, {});

        expect(stdout).toMatchSnapshot();
    });

    it('should error with wrong cli opts', async () => {
        const cmd = "pnpm";
        const args = ['akio', '-a'];

        await expect(() => execa(cmd, args, {})).rejects.toThrow();
    });
});
