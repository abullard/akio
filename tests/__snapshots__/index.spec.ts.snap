// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`index.ts > should run search for build scripts 1`] = `
"
> @abullard/akio@1.0.0 akio /root/Developer/git/abullard/akio
> npx akio -i build

[33mpnpm akio[0m
	-----
1. [35mbuild     [0m â€” 
2. [35mbuild:dev [0m â€” local dev build
3. [35mbuild:prod[0m â€”  build production fake"
`;

exports[`index.ts > should run test:stub npm script 1`] = `
"
> @abullard/akio@1.0.0 akio /root/Developer/git/abullard/akio
> npx akio test\\:stub

[33mpnpm akio[0m
	-----
1. [35mtest:stub [0m â€” 

[32mRun command number?[0m 


> @abullard/akio@1.0.0 test:stub /root/Developer/git/abullard/akio
> node -e "console.log('text')"

text"
`;
